MediaBrowser.WebDashboard/Api/DashboardService.cs

```diff
@@ -111,24 +114,38 @@ public class DashboardService : IService, IRequiresRequest
         /// </summary>
         /// <param name="logger">The logger.</param>
         /// <param name="appHost">The application host.</param>
+        /// <param name="appConfig">The application configuration.</param>
         /// <param name="resourceFileManager">The resource file manager.</param>
         /// <param name="serverConfigurationManager">The server configuration manager.</param>
         /// <param name="fileSystem">The file system.</param>
         /// <param name="resultFactory">The result factory.</param>
         public DashboardService(
             ILogger<DashboardService> logger,
             IServerApplicationHost appHost,
+            IConfiguration appConfig,
             IResourceFileManager resourceFileManager,
             IServerConfigurationManager serverConfigurationManager,
             IFileSystem fileSystem,
             IHttpResultFactory resultFactory)
         {
             _logger = logger;
             _appHost = appHost;
+            _appConfig = appConfig;
             _resourceFileManager = resourceFileManager;
             _serverConfigurationManager = serverConfigurationManager;
             _fileSystem = fileSystem;
             _resultFactory = resultFactory;
+
+            // Validate web content path
+            string webContentPath = DashboardUIPath;
+            bool webContentPathValid = appConfig.NoWebContent() || (Directory.Exists(webContentPath) && Directory.GetFiles(webContentPath).Any());
+            if (!webContentPathValid)
+            {
+                throw new InvalidOperationException(
+                    "The server is expected to host web content, but the provided content directory is either " +
+                    $"invalid or empty: {webContentPath}. If you do not want to host web content with the server, " +
+                    $"you may set the '{Controller.Extensions.ConfigurationExtensions.NoWebContentKey}' flag.");
```

Comment:
Unfortunately throwing the exception here does not halt the application startup. For some reason we are allowing service constructions to fail at app startup by just logging an error message an continuing.

However, I think this is the most logical place to put this check so I have left it here anyways. We should probably update app startup in a separate PR to stop ignoring failures in service startup, particularly for core services like this one.

Reply:
We should maybe add a flag to services that determines whenever or not it's optional.

Reply:
I actually looked into this further already and forgot to write an update here. This service isn't supposed to be instantiated at startup at all, the web API services are created and disposed for each web request. I have opened #2637 to address this and prevent instantiating these services unnecessarily at startup.

I just pushed another change to move this check out of this service and into the server startup in `Program.cs`